File: dotnet/Samples/LanguageDetection/File/EdgeFileLanguageDetectionService.cs
using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.LanguageDetection.Edge.Request;

namespace LanguageWeaver.Sdk.Samples.LanguageDetection.File;

public class EdgeFileLanguageDetectionService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();
        var edgeFileLanguageDetectionRequest = new EdgeFileLanguageDetectionRequest()
        {
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt"),
            InputFormat = Format.Plain,
            Encoding = "UTF-8"
        };

        var edgeLanguageDetectionResult = lwClient.DetectLanguage(edgeFileLanguageDetectionRequest);

        edgeLanguageDetectionResult.Languages
            .ForEach(language => Console.WriteLine(language.Name));
    }
}

File: javascript/backend/samples/language-detection/file/EdgeFileLanguageDetectionService.js
const {
    EdgeLanguageWeaverClient,
    EdgeFileLanguageDetectionRequest,
    Format,
} = require("@language-weaver/lw-sdk-js");
const path = require("path");
const detectEdgeFileLanguage = async () => {
    try{
        const edgeLanguageWeaverClient = await new EdgeLanguageWeaverClient().build();
        const edgeFileLanguageDetectionRequest = new EdgeFileLanguageDetectionRequest();
        edgeFileLanguageDetectionRequest.inputFile = path.resolve("resources/input/input1.txt");
        edgeFileLanguageDetectionRequest.inputFormat = Format.PLAIN;

        const edgeFileLanguageDetectionResult = await edgeLanguageWeaverClient.detectEdgeFileLanguage(edgeFileLanguageDetectionRequest);
        console.log(edgeFileLanguageDetectionResult);
    } catch (e) {
        console.log(e);
    }
}

detectEdgeFileLanguage();